lib/features/login/domain/extensions/oidc_configuration_extensions.dart

```diff
@@ -25,7 +29,29 @@ extension OidcConfigurationExtensions on OIDCConfiguration {
         return '${AppConfig.domainRedirectUrl}/$logoutRedirectOidcWeb';
       }
     } else {
-      return redirectOidcMobile;
-    }
+      return authority == AppConfig.saasRegistrationUrl
+        ? OIDCConstant.twakeWorkplaceRedirectUrl
+        : redirectOidcMobile;
   }
+  }
+
+  String get signInTWPUrl => ServicePath(authority)
+    .withQueryParameters([
+      StringQueryParameter(
+        OIDCConstant.postLoginRedirectUrlPathParams,
+        OIDCConstant.twakeWorkplaceRedirectUrl,
+      ),
+      StringQueryParameter('app', OIDCConstant.appParameter),
+    ])
+    .generateEndpointPath();
+
+  String get signUpTWPUrl => ServicePath(authority)
+    .withQueryParameters([
+      StringQueryParameter(
+        OIDCConstant.postRegisteredRedirectUrlPathParams,
+        OIDCConstant.twakeWorkplaceRedirectUrl,
+      ),
+      StringQueryParameter('app', OIDCConstant.appParameter),
+    ])
+    .generateEndpointPath();
```

Comment:
should have a test to document the form of this URL

Reply:
done